src/pixi_devenv/project.py

```diff
@@ -0,0 +1,322 @@
+from __future__ import annotations
+
+from collections.abc import Iterator
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Union, Any, NewType, assert_never
+
+import serde.toml
+
+
+class DevEnvError(Exception):
+    """Errors raised explicitly by pixi-devenv."""
```

Comment:
Something to consider soon is to have custom exceptions for different error types (like `conda` does: https://docs.conda.io/projects/conda/en/stable/dev-guide/api/conda/exceptions/index.html), as this would break the API (but make it more powerful), and if it's done now it's little work.

Reply:
I agree, but we can extend this later if needed by just creating new subclasses for specific errors, leaving `DevEnvError` to be the top-level error type:

```python
class ResolvePackageNotFound(DevEnvError):
    """..."""
```

This will not be a breaking change.

So I would prefer to create new exception types if we see the need for them in practice, but not before to avoid creating unnecessary code bloat. :+1: